src/subscription_manager/cli_command/register.py

```diff
@@ -439,10 +439,12 @@ def _prompt_for_environment(self):
             )
 
         if self.cp.has_capability(MULTI_ENV):
-            environment = input(_("Environments: ")).replace(" ", "")
+            environment = input(_("Environments: "))
         else:
-            environment = input(_("Environment: ")).strip()
+            environment = input(_("Environment: "))
         readline.clear_history()
+        # ensure the input is not empty
+        environment.strip()
```

Comment:
This line needs to be changed to `environment = environment.strip()` to have any effect, no?
Otherwise a user input of only blank spaces will not trigger a re-prompt for the environment.

Reply:
hm right, missed in the last version of the patch - fixed, thanks